From e0d6b347d8cb61d1e19a35be0536973dd5df4026 Mon Sep 17 00:00:00 2001
From: Lionel Debieve <lionel.debieve@foss.st.com>
Date: Wed, 8 Dec 2021 14:51:21 +0100
Subject: [PATCH 0520/1695] plat-stm32mp1: implement stm32mp_stgen_base()

Implement stm32mp_stgen_base() to get STGEN IOMEM base address.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 3ac640ac431c91cbbd48b5ca92d1ef979fa5ee59 plat-stm32mp1: implement stm32mp_stgen_base()]
Change-Id: Icca93ff1385345114548c67f15e95626031d41d4
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/358363
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/main.c       | 7 +++++++
 core/arch/arm/plat-stm32mp1/stm32_util.h | 2 ++
 2 files changed, 9 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp1/main.c b/core/arch/arm/plat-stm32mp1/main.c
index 001e84da8..77ea562e2 100644
--- a/core/arch/arm/plat-stm32mp1/main.c
+++ b/core/arch/arm/plat-stm32mp1/main.c
@@ -450,6 +450,13 @@ vaddr_t stm32mp_bkpsram_base(void)
 	return io_pa_or_va(&base, BKPSRAM_SIZE);
 }
 
+vaddr_t stm32mp_stgen_base(void)
+{
+	struct io_pa_va base = { .pa = STGEN_BASE };
+
+	return io_pa_or_va(&base, 1);
+}
+
 static bool __maybe_unused bank_is_valid(unsigned int bank)
 {
 	if (IS_ENABLED(CFG_STM32MP15))
diff --git a/core/arch/arm/plat-stm32mp1/stm32_util.h b/core/arch/arm/plat-stm32mp1/stm32_util.h
index 1d58d39fb..4488fcec3 100644
--- a/core/arch/arm/plat-stm32mp1/stm32_util.h
+++ b/core/arch/arm/plat-stm32mp1/stm32_util.h
@@ -29,6 +29,8 @@ bool stm32mp_supports_second_core(void);
 /* Backup registers and RAM utils */
 vaddr_t stm32mp_bkpreg(unsigned int idx);
 vaddr_t stm32mp_bkpsram_base(void);
+/* Platform util for the STGEN driver */
+vaddr_t stm32mp_stgen_base(void);
 
 /* Get device ID from SYSCFG registers */
 uint32_t stm32mp_syscfg_get_chip_dev_id(void);
-- 
2.47.2

